log.go

```diff
@@ -87,6 +91,15 @@ func (el *eventLogger) EventBegin(ctx context.Context, event string, metadata ..
 		}))
 
 		el.Event(ctx, event, metadata...)
+		span.SetTag("system", el.system)
+		//This feels a bit clunky
+		for k := range metadata {
```

Comment:
You can at least avoid the additional lookup by using:

```go
for _, m := range metadata {
    for l, v := range m.Loggable() {
        span.SetTag(l, v)
    }
}
```

But still...

Reply:
This all should actually be done inside the `Event` method so we can avoid doing it if the event logger is not enabled.

Reply:
Could we not check `WriterGroup.Active()` (and return an entirely blank event if it's not active)?

Also, this metadata should probably be stored using `LogFields`? I believe "tags" are for discrete things (that should be indexed).

Reply:
I like the simplicity of performing the `WriterGroup.Active()` check in the events `Done` function. However I think a new subcommand is needed on `ipfs log` to set the `WriterGroup` as active for the case when we want to collect traces. Otherwise we would need to run `ipfs log tail` before traces are sent to the collector. I have open an issue [here](https://github.com/ipfs/go-ipfs/issues/4487) to discuss it.